server/src/main/java/org/elasticsearch/index/IndexSettings.java

```diff
@@ -243,9 +243,11 @@
 
     /**
      * Specifies if the index should use soft-delete instead of hard-delete for update/delete operations.
+     * Soft-deletes is enabled by default for 7.0+ indices.
      */
-    public static final Setting<Boolean> INDEX_SOFT_DELETES_SETTING =
-        Setting.boolSetting("index.soft_deletes.enabled", true, Property.IndexScope, Property.Final);
+    public static final Setting<Boolean> INDEX_SOFT_DELETES_SETTING = Setting.boolSetting("index.soft_deletes.enabled",
+        settings -> Boolean.toString(IndexMetaData.SETTING_INDEX_VERSION_CREATED.get(settings).onOrAfter(Version.V_7_0_0)),
```

Comment:
I wonder if all unit tests are setting this correctly. If the setting is absent, we could perhaps assume that soft-deletes are enabled, instead of disabled.

Reply:
if we create IndexMetadata without this setting I think it will barf. so I think we are good.